firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleReader.java

```diff
@@ -185,9 +186,12 @@ private String readJsonString(int length) throws IOException {
    */
   private boolean pullMoreData() throws IOException {
     buffer.compact();
-    int read = dataReader.read(buffer);
+    int bytesToRead = Math.min(bundleInputStream.available(), buffer.remaining());
+    byte[] bytes = new byte[bytesToRead];
```

Comment:
s/Consider reading/Please read

Reply:
Similarly, I feel better to avoid performance optimization without proof. The end result is some byte counting, which is bad for readability.

The other place we do this is acceptable, because there is no good alternative..

Reply:
You are doubling the amount of memory consumed here. Gil pushed back heavily against this in the original PR.

Reply:
I still think this is premature..but done anyways.